CopilotKit/packages/backend/src/lib/openai-assistant-adapter.ts

```diff
@@ -177,6 +184,77 @@ export class OpenAIAssistantAdapter implements CopilotKitServiceAdapter {
       };
     }
   }
+
+  async buildInstructions(forwardMessages: Message[]): Promise<string> {
+    const instructions = forwardMessages
+      .filter((message) => message.role === "system")
+      .map((message) => message.content)
+      .join("\n\n");
+  
+    return instructions;
+  }
+
+  async buildTools(forwardedProps: any): Promise<any[]> {
+    const tools = [
+      ...(forwardedProps.tools || []), 
+      ...(this.codeInterpreterEnabled ? [{ type: "code_interpreter" }] : []),
+      ...(this.retrievalEnabled ? [{ type: "retrieval" }] : []),
+    ];
+  
+    return tools;
+  }
+
+  async handleStreamingRun(run: OpenAI.Beta.Threads.Runs.RunStream) {
+    return new Promise((resolve) => {
```

Comment:
this implementation is on the right direction, but I believe we want to use our existing abstractions here,

see `writeChatCompletionChunk`, `writeChatCompletionEnd` and `AssistantSingleChunkReadableStream` for an example.

@mme can you chime in here please?